src/metabase/driver/postgres.clj

```diff
@@ -322,65 +322,43 @@
 
 (defmethod sql.qp/datetime-diff [:postgres :year]
   [_driver _unit x y]
-  (hx/cast
-   :integer
-   (hsql/call
-    :extract
-    :year
-    (hsql/call
-     :age
-     (date-trunc :day y)
-     (date-trunc :day x)))))
+  (let [interval (hsql/call :age (date-trunc :day y) (date-trunc :day x))]
+    (hx/->integer (hsql/call :extract :year interval))))
```

Comment:
There's nothing new in the changes to postgres, just compressing things a bit and naming when expressions have meanings